sources/api/schnauzer/src/helpers/mod.rs

```diff
@@ -563,6 +570,123 @@ pub fn host(
+    let template_name = template_name(renderctx);
+    check_param_count(helper, template_name, 1)?;
+
+    let cidr_param = get_param(helper, 0)?;
+    let cidr = cidr_param
+        .as_str()
+        .with_context(|| error::InvalidTemplateValueSnafu {
+            expected: "string",
+            value: cidr_param.to_owned(),
+            template: template_name,
+        })?;
+
+    let ip = validate_and_parse_cidr(cidr)?;
+    out.write(&ip.to_string())
+        .with_context(|_| error::TemplateWriteSnafu {
+            template: template_name.to_owned(),
+        })?;
+
+    Ok(())
+}
+
+/// Replaces a specific octet in an IPv4 address with a new value. The octet is specified by index (0-3).
+pub fn replace_ipv4_octet(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    
+    trace!("Starting replace_ipv4_octet helper");
+    let template_name = template_name(renderctx);
+    check_param_count(helper, template_name, 3)?;
+
+    let ip_param = get_param(helper, 0)?;
+    let octet_index_param = get_param(helper, 1)?;
+    let new_value_param = get_param(helper, 2)?;
+
+    let ip_str = ip_param
+        .as_str()
+        .with_context(|| error::InvalidTemplateValueSnafu {
+            expected: "string",
+            value: ip_param.to_owned(),
+            template: template_name,
+        })?;
+    let octet_index = octet_index_param
+        .as_u64()
+        .with_context(|| error::InvalidTemplateValueSnafu {
+            expected: "integer",
+            value: octet_index_param.to_owned(),
+            template: template_name,
+        })? as usize;
+    let new_value = new_value_param
+        .as_str()
+        .with_context(|| error::InvalidTemplateValueSnafu {
+            expected: "string",
+            value: new_value_param.to_owned(),
+            template: template_name,
+        })?;
+        
+    let ip = Ipv4Addr::from_str(ip_str).map_err(|error| {
+        RenderError::from(error::TemplateHelperError::InvalidIPAddress {
+            ip_address: ip_str.to_string(),
+            source: error,
+        })
+    })?;
+
+    let mut octets = ip.octets();
+    if octet_index > 3 {
+        return Err(RenderError::new(format!(
+            "Invalid index {} for IP address {}",
+            octet_index, ip_str
+        )));
+    }
+
+    octets[octet_index] = new_value
+        .parse::<u8>()
+        .map_err(|_| RenderError::new("Failed to parse the new octet value as an integer"))?;
+
+    let new_ip = Ipv4Addr::from(octets).to_string();
+    out.write(&new_ip).with_context(|_| error::TemplateWriteSnafu {
+        template: template_name.to_owned(),
+    })?;
+    
+    Ok(())
+}
+
+fn validate_and_parse_cidr(cidr: &str) -> Result<IpAddr, RenderError> {
+    let parsed_cidr: AnyIpCidr = cidr.parse().map_err(|_| error::InvalidCidrSnafu { cidr: cidr.to_string() }.build())?;
```

Comment:
If you mark `source` as [`NetworkParseError`](https://docs.rs/cidr/latest/cidr/errors/enum.NetworkParseError.html) in `InvalidCidrSnafu` you could just use `context` here, and also add additional information to the error message about why the parse failed.